tools/rebuild_provision_request.rb

```diff
@@ -0,0 +1,166 @@
+
+  response
+end
+
+def ws_only_files(prov_options, response)
+  response["additional_values"] = prov_options.delete(:ws_values)
+  response["ems_custom_attributes"] = prov_options.delete(:ws_ems_custom_attributes)
+  response["miq_custom_attributes"]  = prov_options.delete(:ws_miq_custom_attributes)
+end
+
+def dialog_field_values(dialog_tab, prov_options)
+  dialog_tab[:fields].keys.each do |field|
+    value = prov_options[field]
+    value = value.first if value.kind_of?(Array)
+    yield(field.to_s, value) if value.present?
+    prov_options.delete(field)
+  end
+end
+
+def process_dialog_tab(tab, dialog_tab, prov_options, response)
+  puts "Processing tab: #{tab}"
+  tab_name = respond_to?(tab, true) ? tab : :general_tab
+  send(tab_name, tab, dialog_tab, prov_options, response)
+end
+
+def requester(_tab, dialog_tab, prov_options, response)
+  request = response["requester"]
+  dialog_field_values(dialog_tab, prov_options) do |field, value|
+    request[field] = value
+  end
+end
+
+def purpose(_tab, _dialog_tab, prov_options, response)
+  request = response["tags"]
+  tag_ids = prov_options.delete(:vm_tags)
+  Classification.where(:id => tag_ids).each do |tag|
+    request[tag.parent.name] = tag.name
+  end
+end
+
+def environment(_tab, dialog_tab, prov_options, response)
+  request = response["vm_fields"]
+  if prov_options[:placement_auto].first == true
+    request["placement_auto"] = true
+    return
+  end
+
+  request["placement_auto"] = false
+end
+
+def general_tab(_tab, dialog_tab, prov_options, response)
+  request = response["vm_fields"]
+  dialog_field_values(dialog_tab, prov_options) do |field, value|
+    request[field] = value
+  end
+end
+
+def customize(tab, dialog_tab, prov_options, response)
+  prov_options.delete(:customization_template_script)
+  general_tab(tab, dialog_tab, prov_options, response)
+end
+
+def schedule(tab, dialog_tab, prov_options, response)
+  if prov_options[:schedule_type].first == "immediately"
+    prov_options.delete(:schedule_type)
+    prov_options.delete(:schedule_time)
+  else
+    prov_options[:schedule_time] = prov_options[:schedule_time].to_s
+  end
+  general_tab(tab, dialog_tab, prov_options, response)
+end
+
+request_id = ARGV[0]
+authed_url = ARGV[1]
+run_it = ARGV[2]
+if request_id.blank?
+  print_usage
+  exit
+end
+
+prov_options   = load_request_options(request_id)
+dialog_content = load_dialog_content(prov_options)
+response       = parse_dialog_tabs(dialog_content, prov_options)
+url = "RestClient.post 'http://#{authed_url}#{URL_PATH}', '#{response.to_json}'"
+
+puts "\nOutput (#{OUTPUT_FORMAT}):"
+if OUTPUT_FORMAT == 'json'
+  require 'json'
+  puts JSON.pretty_generate(response)
+else
+  require 'pp'
+  pp response
+end
+
+if authed_url
+  puts ""
+  puts "RestClient API POST - (to be copied into a Rails console)"
+  puts ""
+  puts url
+end
+
+if run_it
+  puts ""
+  puts "RE Running the Rest API POST to request_id: #{request_id}"
+  result = eval(url)
```

Comment:
@syncrou Looks like both me and rubo👮  are not a fan of `eval`.  Would be better to rework this to avoid that call.